### PR TITLE
Support viewportWidth for pattern

### LINKED ISSUE #22147: Allow patterns to declare a "preferred" viewport for previews
Right now we set an arbitrary width for pattern previews, and attempt to scale things to fit. This handles multiple cases somewhat well enough. However, for some specific designs, it'd be desirable if the pattern could declare their preferred viewing width to avoid visual breakage. I don't expect most patterns to use this, but some could benefit from it.

Examples from @iamtakashi

image (6) (6)

### PR DESCRIPTION
**Pull request by manooweb**
Fixes #22147 

Add support of viewportWidth for each pattern.

**Test:**

- open pattern inserter
- check two buttons pattern width
- the pattern width must to be larger than on master branch

### CODE REVIEW

File: lib/patterns/two-buttons.php [hunk 1]
```diff
@@ -6,6 +6,7 @@
...
+	'title'         => __( 'Two Buttons', 'gutenberg' ),
+	'content'       => "<!-- wp:buttons {\"align\":\"center\"} -->\n<div class=\"wp-block-buttons aligncenter\"><!-- wp:button {\"backgroundColor\":\"very-dark-gray\",\"borderRadius\":0} -->\n<div class=\"wp-block-button\"><a class=\"wp-block-button__link has-background has-very-dark-gray-background-color no-border-radius\">Button One</a></div>\n<!-- /wp:button -->\n\n<!-- wp:button {\"textColor\":\"very-dark-gray\",\"borderRadius\":0,\"className\":\"is-style-outline\"} -->\n<div class=\"wp-block-button is-style-outline\"><a class=\"wp-block-button__link has-text-color has-very-dark-gray-color no-border-radius\">Button Two</a></div>\n<!-- /wp:button --></div>\n<!-- /wp:buttons -->",
+	'viewportWidth' => 500,
```
Comment (youknowriad) [inline lib/patterns/two-buttons.php:11]: Any thoughts on the property name @mtias @enriquesanchez ?

Comment (mtias) [reply lib/patterns/two-buttons.php:11]: Maybe `previewWidth`?

Comment (youknowriad) [reply lib/patterns/two-buttons.php:11]: i don't feel strongly either way personally but some clarifications here:

 - The prop for block preview is named `viewportWidth` which i believe is the reason this was chosen here.
 - This is not the "preview width" as the preview width is stable (inserter width) but the base width used before scaling the preview down. `previewWidth` could work though.

Comment (youknowriad) [reply lib/patterns/two-buttons.php:11]: @manooweb can we update this to "previewWidth" and ship?

Comment (mtias) [reply lib/patterns/two-buttons.php:11]: If you think we should remain consistent with the prop name, that's also fine.

Comment (youknowriad) [reply lib/patterns/two-buttons.php:11]: I'm on the fence personally. I'm going to merge and we'll see. 

Also cc @aduth as my go to person for naming things :P

**Review by youknowriad — APPROVED** [review]

File: lib/patterns/two-buttons.php (hunk 1 above)
Comment (aduth) [reply lib/patterns/two-buttons.php:11]: If I understand correctly, it's a bit strange to me to consider preview configurations as top-level properties of a pattern, especially when not notated as such. Is there a purpose for `viewportWidth` outside just the preview? Do we foresee needing to have any additional configuration for preview?

Wondering if it's something where `preview` could be a nested object of settings relevant for the preview.

`preview.width`, `preview.viewportWidth`, etc.

Comment (youknowriad) [reply lib/patterns/two-buttons.php:11]: The actual preview width doesn't really matter, it's in general dependent on the wrapper but in order to show a block preview (or a template preview), we need to define a width for the canvas (editor) and  then scale it down.

**Pull request merged**